src/sync/incomingSyncUtils.js

```diff
@@ -500,11 +500,11 @@ export const createOrUpdateRecord = (database, settings, recordType, record) =>
       break;
     }
     case 'Report': {
-      const { ID: id, title, type, data } = record;
+      const { ID: id, title, type, json } = record;
       try {
-        const parsedData = JSON.parse(data);
+        const parsedData = JSON.parse(json);
         const shouldSetData = checkIsObject(parsedData);
-        internalRecord = { id, title, type, _data: shouldSetData ? data : null };
+        internalRecord = { id, title, type, _data: shouldSetData ? json.data : null };
```

Comment:
sorry my mistake - `json` will actually be a string here. So, we will need to parse the string - for example `parsedData.data`. BUT we also need to make sure we are setting a STRING as `_data`, NOT an object!

Reply:
Ok, so apart from checking `parsedData.data` we can add something like `checkIsString` to check`json` as well, and if it's all good we assigned that to `_data` (or `_json`)
Am I right?

Reply:
Hmm it does get a bit confusing. Consider this simplified case, where I think `testData` is a simple version of the `json` field and `testOne` is `json.data`:

```
const testData = { testOne: [{ testTwo: 2 }] };
const testDataAsString = JSON.stringify(testData);
const parsedTestData = JSON.parse(testDataAsString);
console.log(typeof parsedTestData.testOne);
```

then what gets logged?

Reply:
haha I guess we will see the original `testData.testOne`, but with a couple of shakings, right?

Reply:
Great, thanks! I reckon it's solved here: 01ad0a9